rfc/rfc004-distributed-document-format.md

```diff
@@ -0,0 +1,143 @@
+
+In addition to the registered header parameters, the following headers MUST be present as protected headers:
+* **sigt**: (signing time) MUST contain the signing time of the document in UTC as string, formatted according to [RFC3339](https://tools.ietf.org/html/rfc3339).
+* **version**: MUST contain the format version of the document as number. For this version of the format the version MUST be 1.
+* **prevs**: (previous documents) MUST contain the references (see section 3.2) of the preceding documents (see section 3.4).
+
+The following headers MAY be present as protected headers (see section 3.4 for details):
+* **tid**: (timeline ID) MUST contain the reference to the document that started the timeline.
+* **tiv**: (timeline version) MUST contain a numeric version indicating the version of the document on the timeline.
+
+### 3.2. Document Reference
+The document reference uniquely identifies a document and is used to refer to it. It MUST be calculated by
+taking the bytes of the JWS EXACTLY as received and hashing it using SHA-1.
+
+When serializing a reference to string form it MUST be hexadecimal encoded and SHOULD be lowercase.
+
+Example:
+
+```148b3f9b46787220b1eeb0fc483776beef0c2b3e```
+
+### 3.3. Ordering, branching and merging
+Documents MUST form a chain by referring to the previous document. This MAY be used to establish *casual ordering*, e.g.
+registration of a care organization as child object of a vendor. A new document MUST be appended to the end of the chain
+by referring to the last document of the chain (*head*) by including its reference in the **prevs** field.
+
+Since it takes some time for the documents to be synced to all network peers (eventual consistency) there COULD be
+multiple documents referring to the previous documents in the **prevs** field, a phenomenon called *branching*. Since
+branches (especially old and/or long ones) may cause documents to be reordered which hurts performance they MUST be
+merged as soon as possible. Branches are merged by specifying their heads in the **prevs** field:
+
+![RFC structure](.gitbook/assets/rfc004-branching.svg)
+
+The first document in the chain is the *genesis document* and SHALL NOT have any **prevs** entries. There MUST only be
+one genesis document for a network and subsequent genesis documents MUST be ignored.
+
+When processing a chain the system MUST start at the genesis document and work its way to the head(s) processing subsequent
+documents. When encountering a branch the documents on all branches from that point up until the merge
+MUST be ordered before processing. Individual ordering of documents happens by hash; lower hashes are processed first.
+For example in the diagram above, the processing order of documents is `A -> B -> C -> D -> E -> F`.
+
+TODO: Does this ordering between hashes actually matter? Since when ordering is required documents have to be in the
```

Comment:
nope, ordering of hashes is only required for the `[head]` list to get to a single hash value.